### DISCUSSION TITLE
Not making on the demanded cores

### DISCUSSION THREAD

**Discussion by AmbroiseJuston** [idaholab/moose#26719] [Q&A General]
Goof morning, I am running  program I know works on a new machine (48  physical cores, 96 total virtual cores). When running the make -j80 and then running the executable i get this output below:

```pre
(moose) [moose@lprc-b72-112-fem xenon]$ make -j80
(moose) [moose@lprc-b72-112-fem xenon]$ ./run_tests -j80
test:kernels/simple_diffusion.test ........................................................................ OK
--------------------------------------------------------------------------------------------------------------
Ran 1 tests in 0.3 seconds. Average test time 0.2 seconds, maximum test time 0.2 seconds.
1 passed, 0 skipped, 0 pending, 0 failed
(moose) [moose@lprc-b72-112-fem xenon]$ make -j80
(moose) [moose@lprc-b72-112-fem xenon]$ cd ~/projects/xenon
(moose) [moose@lprc-b72-112-fem xenon]$ cd ~/projects/xenon/problems
(moose) [moose@lprc-b72-112-fem problems]$ ls
xenon_input.i
(moose) [moose@lprc-b72-112-fem problems]$ ../xenon-opt -i xenon_input.i
hwloc/linux: Ignoring PCI device with non-16bit domain.
Pass --enable-32bits-pci-domain to configure to support such devices
(warning: it would break the library ABI, don't enable unless really needed).
Framework Information:
MOOSE Version:           git commit ed55c164c2 on 2024-01-17
LibMesh Version:         
PETSc Version:           3.20.1
SLEPc Version:           3.20.0
Current Time:            Fri Feb  2 10:28:18 2024
Executable Timestamp:    Fri Feb  2 10:14:01 2024

Parallelism:
  Num Processors:          1
  Num Threads:             1
```

What i noticed was that it was specifying that only 1 processor and thread was being used for the program, when I obviously want 80, (which is what i though i specified). Any help on what is happening or if MOOSE differentiates between physical and virtual cores would be veryappreiciated. When running a stress test, i can see the cores being used so i know that they are available.

**Comment by GiudGiud** [comment]
Hello

There a few things you did there.

First make -j80 correctly used 80 processes to build the code. It seems it was already built so it just skipped. But next time you start from afresh it will use those resources. 

Then ./run_tests -j80 allows for 80 tests to run in parallel. This is not the same as using 80 parallel processes to run each test. What you did is faster though, but we don't expect much gain between -j10 and -j80 because of limitations in the Python threading library. 

Finally you ran an input file in serial, with one process and one thread. 

If you want to run that same input file in parallel, you can use
mpirun -n 80 executable-opt -i inputfile.i

**Comment by milljm** [comment]
In order to make your test use 80 cores, you'll need to modify that test to do so:

```conf
[Tests]
  [test]
    type = 'Exodiff'
    input = 'simple_diffusion.i'
    exodiff = 'simple_diffusion_out.e'

    issues = '#1493'
    design = 'kernels/Diffusion.md'
    requirement = 'The system shall run a simple 2D linear diffusion problem with Dirichlet boundary conditions on a regular mesh.'
    min_parallel = 80
  []
[]
```
_min_parallel = 80_

You can also force the TestHarness to throw parallel options at your binary for every test with:
```bash
./run_tests  --parallel 8 -v
<trimmed>
kernels/simple_diffusion.test:   Num Processors:          8
kernels/simple_diffusion.test:   Num Threads:             1
```
